### PR TITLE
fix: Table reordering fixed

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

After the merge of the #5056 the table reordering had an issue on the first render where the elements didn't have the `widthData` properly set (it was relying on a `useEffect` hook that was removed). 

## Test Plan

Go to /det/projects/ and "play" with the table 😁

known features

- [x] Row selection
- [x] Column selection
- [x] Reorder columns (D&D)
- [x] Filter
- [x] Resize column width
- [x] Menu with right click
- [x] Sort
- [x] Pagination
- [x] All row height should be the same (for pin feature)

## Commentary (optional)

demo:

https://user-images.githubusercontent.com/104855841/193300868-2f1d770d-9995-4af2-8977-cbb27dd4beff.mp4

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keita-determined — COMMENTED** [review]
Thanks for the fix
The reorder feat is fixed but there's another regression. 
The table header is broken after adding more columns. Cannot scroll horizontally 

https://user-images.githubusercontent.com/109113616/193321398-fd6e9b5a-8c35-41f5-b523-1a8701560fcc.mov

**Comment by keita-determined** [comment]
The default width is broken.
Some column widths are too short, some are too long

https://user-images.githubusercontent.com/109113616/193634718-656267e7-7980-404c-a5f8-3e80be8c0d07.mov

**Comment by keita-determined** [comment]
[The previous issue](https://github.com/determined-ai/determined/pull/5159#pullrequestreview-1127123690) is fixed.
However, I see another issue. It works on latest-master website, but not in this PR.

I cannot resize the column width with few columns. I think it's the last bug so far (hopefully :))

https://user-images.githubusercontent.com/109113616/193644407-3f54d8e7-6c24-468c-875a-104c378d0418.mov

File: webui/react/src/components/InteractiveTable.tsx [hunk 1]
```diff
@@ -406,6 +396,33 @@ const InteractiveTable: InteractiveTable = ({
...
+
+  const [isResizing, setIsResizing] = useState(false);
+
+  const { dragState } = useDragLayer((monitor) => {
+    const deltaX = monitor.getDifferenceFromInitialOffset()?.x;
+    const dragState = deltaX ? (deltaX > 0 ? 'draggingRight' : 'draggingLeft') : 'notDragging';
+    return { dragState };
+  });
+
+  const spinning = !!(loading as SpinProps)?.spinning || loading === true;
```
Comment (keita-determined) [inline webui/react/src/components/InteractiveTable.tsx:424]: just `loading` should be fine

File: webui/react/src/components/InteractiveTable.tsx [hunk 2]
```diff
@@ -406,6 +396,33 @@ const InteractiveTable: InteractiveTable = ({
...
+      dropRightStyles:
+        widths.map((width, idx) => ({
+          left: `${width / 2}px`,
+          width: `${(width + (widths[idx + 1] ?? WIDGET_COLUMN_WIDTH)) / 2}px`,
+        })) ?? [],
+      widths: widths ?? [],
+    };
+  });
+
+  const [isResizing, setIsResizing] = useState(false);
```
Comment (keita-determined) [inline webui/react/src/components/InteractiveTable.tsx:416]: nice to have type <boolean>

File: webui/react/src/components/InteractiveTable.tsx [hunk 3]
```diff
@@ -406,6 +396,33 @@ const InteractiveTable: InteractiveTable = ({
     [pageWidth],
   );
 
+  const [widthData, setWidthData] = useState(() => {
```
Comment (keita-determined) [inline webui/react/src/components/InteractiveTable.tsx:399]: nice to have type

File: webui/react/src/components/InteractiveTable.tsx [hunk 4]
```diff
@@ -267,12 +268,16 @@ const HeaderCell = ({
       const dragItem = monitor.getItem() || {};
       const dragIndex = dragItem?.index;
       const deltaX = monitor.getDifferenceFromInitialOffset()?.x;
-      const dragState = deltaX ? (deltaX > 0 ? 'draggingRight' : 'draggingLeft') : 'notDragging';
+      const internalDragState = deltaX
```
Comment (keita-determined) [inline webui/react/src/components/InteractiveTable.tsx:271]: Too many ternary operators aren't good for readability

File: webui/react/src/components/Transfer.tsx [hunk 5]
```diff
@@ -194,7 +194,13 @@ const Transfer: React.FC<Props> = ({
...
             {renderVisibleRow}
           </List>
-          <Button type="link" onClick={() => moveToLeft(filteredVisibleEntries)}>
+          <Button
+            type="link"
+            onClick={() => {
+              moveToLeft(filteredVisibleEntries);
+              // removing everything was keeping the columns out of sync with the UI...
+              moveToRight(['id', 'name']);
+            }}>
```
Comment (keita-determined) [inline webui/react/src/components/Transfer.tsx:203]: Probably its not a good idea to keep `id` and `name` cuz user might think these are default columns, but we can actually get rid of them. Also, it fails test cases.

**Review by keita-determined — COMMENTED** [review]
Just one thing about test cases

**Review by ClaireNeveu — APPROVED** [review]
Discussed offline about making the test plan more explicit. Approved for now given the extensive QA done by Keita on the PR but in the future test plans should be more explicit than this.

**Pull request merged**